docs/developer-docs/backend/rust/message-inspect.mdx

```diff
@@ -0,0 +1,55 @@
+---
+keywords: [intermediate, rust, tutorial, message inspection]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Message inspection
+
+<MarkdownChipRow labels={["Intermediate", "Rust", "Tutorial"]} />
+
+## Overview
+
+A canister can inspect an ingress message and either accept or decline the message through the canister's HTTP interface. If the message is accepted, the canister will execute it.
+
+A canister that does not have a Wasm module installed will reject any ingress messages. If the canister does not implement `canister_inspect_message`, all ingress messages will be accepted.
+
+The `canister_inspect_message` functionality should not be used for definitive access control. This is because it is executed by a single replica, without full consensus, and its result could be spoofed by a malicious boundary node. However, since it does not go through full consensus, a malicious user cannot drain cycles from your canister because you don’t pay the cost for the message going through consensus.
+
+Using the `canister_inspect_message` endpoint, the canister can invoke `ic0.accept_message : () → ()` to accept the message.
+
+You can learn more in the [IC interface specification](/docs/current/references/ic-interface-spec/#ingress-message-inspection).
+
+This guide will describe how to use `inspect_message` within a Rust canister.
+
+## `inspect_message`
+
+The `canister_inspect_message` entrypoint can be defined and exported by a Rust canister using `#[inspect_message]`.
+
+The function within this attribute cannot have a return value, and a canister can only have one `canister_inspect_message` entrypoint.
+
+```rust
+#[inspect_message]
+fn inspect_message_function() {
+    // ...
+}
+```
+
+## `inspec_message` with `guards`
+
+To obtain the ingress parameters while implementing `canister_inspect_message`, you must implement a `guard`. A guard is for rejecting a message. Any reply from non-guard code is either a successful reply or a trap.
+
+You can implement a guard in your code such as:
+
+```rust
+#[update(guard = "can_prepare")]
+#[candid_method(update)]
+fn create_batch() -> CreateBatchResponse {
+    STATE.with(|s| match s.borrow_mut().create_batch(time()) {
+        Ok(batch_id) => CreateBatchResponse { batch_id },
+        Err(msg) => trap(&msg),
+    })
+}
+```
+
+
```

Comment:
```suggestion
```

The `guard` function in `ic_cdk` has no relationship with the `canister_inspect_message`.

We should remove this section